ml4gw/types.py

```diff
@@ -1,10 +1,25 @@
-from torchtyping import TensorType
-
-WaveformTensor = TensorType["batch", "num_ifos", "time"]
-PSDTensor = TensorType["num_ifos", "frequency"]
-ScalarTensor = TensorType["batch"]
-VectorGeometry = TensorType["batch", "space"]
-TensorGeometry = TensorType["batch", "space", "space"]
-NetworkVertices = TensorType["num_ifos", 3]
-NetworkDetectorTensors = TensorType["num_ifos", 3, 3]
-TimeSeriesTensor = TensorType["num_channels", "time"]
+from typing import Union
+
+from jaxtyping import Float
+from torch import Tensor
+
+WaveformTensor = Float[Tensor, "batch num_ifos time"]
+PSDTensor = Float[Tensor, "num_ifos frequency"]
+ScalarTensor = Float[Tensor, "batch"]
+VectorGeometry = Float[Tensor, "batch space"]
+TensorGeometry = Float[Tensor, "batch space space"]
+NetworkVertices = Float[Tensor, "num_ifos 3"]
+NetworkDetectorTensors = Float[Tensor, "num_ifos 3 3"]
+
+
+TimeSeries1d = Float[Tensor, "time"]
+TimeSeries2d = Float[TimeSeries1d, "channel"]
+TimeSeries3d = Float[TimeSeries2d, "batch"]
+TimeSeries1to3d = Union[TimeSeries1d, TimeSeries2d, TimeSeries3d]
+
+FrequencySeries1d = Float[Tensor, "frequency"]
```

Comment:
Could all of these be succinctly put as 
`FrequencySeries = Float[Tensor, "*batch *channel frequency"]`
Where the `*` means it can refer to 0 or more dimensions

Reply:
https://docs.kidger.site/jaxtyping/api/array/

Reply:
The multiple axes symbols can only be used once per array

Reply:
One of the reasons I ended up being disappointed with `jaxtyping` - having that functionality would be pretty useful, but I found an old issue about it and it sounds like they deliberately don't want that to be possible

Reply:
Oh, and doing `Float[Tensor, "*batch frequency"]` would effectively mean "at least 1d", which would be a problem for some functions that cap at 3d.